config/src/main/java/com/typesafe/config/ConfigParseOptions.java

```diff
@@ -72,15 +72,15 @@ public ConfigParseOptions setSyntax(ConfigSyntax syntax) {
      * @return options with the syntax set
      */
     public ConfigParseOptions setSyntaxFromFilename(String filename) {
-        ConfigSyntax syntax = ConfigImplUtil.syntaxFromExtension(filename);
+        ConfigFormat syntax = ConfigImplUtil.syntaxFromExtension(filename);
         return setSyntax(syntax);
     }
 
     /**
      * Gets the current syntax option, which may be null for "any".
      * @return the current syntax or null
      */
-    public ConfigSyntax getSyntax() {
+    public ConfigFormat getSyntax() {
```

Comment:
So with the ConfigFormat and ConfigSyntax, I made it so that the ConfigSyntax enum implements ConfigFormat.  Therefore a ConfigSyntax _is_ a ConfigFormat. This allows for the least amount of code changes overall in the implementations. I 100% agree on your point on the getter here.